mantidimaging/core/operations/rotate_stack/rotate_stack.py

```diff
@@ -96,15 +95,94 @@ def _update_angle(angle, dropdown):
             angle.setEnabled(False)
 
 
-def _rotate_image_inplace(data, angle=None):
-    data[:, :] = rotate(data[:, :], angle)
+def _get_cardinal_angle(value) -> int:
+    """
+    Calculates the angle based on the input value, by adding 45 to the value modulo 360,
+    dividing it by 90, multiplying by 90, and then subtracting 90 if the value modulo 360 modulo
+    90 is equal to 45, otherwise subtracting 0 to get the cardinal angle the aspect ratio should be
+    rotated by.
+
+    param: value: angle of rotation
+    return: angle: cardinal angle of rotation
+    """
+    angle = ((value % 360 + 45) // 90 * 90 - (90 if (value % 360) % 90 == 45 else 0))
+    return angle % 360
+
+
+def _compute_rotation(data, angle, progress) -> np.ndarray:
```

Comment:
I'd rather we preserved the name `compute` for the functions passed to `run_compute_func`.

It could either be `do_rotation`, or content could be moved into `filter_func`